src/bin/rustfmt.rs

```diff
@@ -306,8 +321,8 @@ fn determine_operation(matches: &Matches) -> FmtResult<Operation> {
             Some(dir)
         });
 
-    // if no file argument is supplied, read from stdin
-    if matches.free.is_empty() {
+    // if no file argument is supplied and `--file-lines` is not specified, read from stdin
+    if matches.free.is_empty() && !matches.opt_present("file-lines") {
```

Comment:
On this note, I'm tempted to make `--file-lines` mutually exclusive with specifying file names on the command line, since it's redundant and in the worst case results in unnecessary parsing. I'd leave that to a later PR though.